### PR TITLE
RCAL-911 & 932: remove units from MOS and ELP pipelines.

### PR DESCRIPTION
**Pull request by mairanteodoro**
Resolves [RCAL-911](https://jira.stsci.edu/browse/RCAL-911)
Resolves [RCAL-932](https://jira.stsci.edu/browse/RCAL-932)

This PR modifies several files so that units are not used anymore. Additionally, the filename used in the regression tests was changed to remove the `ggsaa` component.

The information about units is still retained in the RAD schema.

## Tasks
- [X] **request a review from someone specific**, to avoid making the maintainers review every PR
- [X] add a build milestone, i.e. `24Q4_B15` (use the [latest build](https://github.com/spacetelescope/romancal/milestones) if not sure)
- [ ] Does this PR change user-facing code / API? (if not, label with `no-changelog-entry-needed`)
  - [x] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types)
  - [X] update or add relevant tests
  - [X] update relevant docstrings and / or `docs/` page
  - [ ] [start a regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) and include a link to the running job ([click here for instructions](https://github.com/spacetelescope/RegressionTests/blob/main/docs/running_regression_tests.md))
    - [ ] Do truth files need to be updated ("okified")?
      - [ ] **after the reviewer has approved these changes**, run `okify_regtests` to update the truth files
- [ ] if a JIRA ticket exists, [make sure it is resolved properly](https://github.com/spacetelescope/romancal/wiki/How-to-resolve-JIRA-issues)

<details><summary>news fragment change types...</summary>

  - ``changes/1445.general.rst``: Remove units from romancal.

</details>

### CODE REVIEW

File: romancal/photom/tests/test_photom.py [hunk 1]
```diff
@@ -153,59 +143,59 @@ def test_apply_photom1():
...
-    assert output_model.meta.photometry.pixelarea_steradians.unit == area_ster.unit
+    # assert output_model.meta.photometry.pixelarea_steradians.unit == area_ster.unit
     assert np.isclose(
-        output_model.meta.photometry.pixelarea_arcsecsq.value,
+        output_model.meta.photometry.pixelarea_arcsecsq,
         area_a2.value,
         atol=1.0e-7,
     )
-    assert output_model.meta.photometry.pixelarea_arcsecsq.unit == area_a2.unit
+    # assert output_model.meta.photometry.pixelarea_arcsecsq.unit == area_a2.unit
```
Comment (schlafly) [inline romancal/photom/tests/test_photom.py:156]: Let's just delete these, right?

**Review by schlafly — COMMENTED** [review]
Thanks Mairan.  This looks great.  I left one minor comment about deleting some presently commented-out code.  Please spot check one of the L2 and L3 pipeline outputs for unexpected differences but this looks good to me.

FYI, @ddavis-stsci , this contains two sets of changes that touch all of the regtest files:
- changing the regtest filenames to remove the ggsaa component
- removing all of the units.

File: romancal/skymatch/skystatistics.py [hunk 2]
```diff
@@ -126,14 +126,14 @@ def calc_sky(self, data):
...
 
         """
-        imstat = ImageStats(image=data.value, fields=self._fields, **(self._kwargs))
+        imstat = ImageStats(image=data, fields=self._fields, **(self._kwargs))
         stat = self._skystat(imstat)  # dict or scalar
 
         # re-attach units:
         if hasattr(stat, "__len__"):
-            self.skyval = {k: value * data.unit for k, value in stat.items()}
+            self.skyval = {k: value for k, value in stat.items()}
```
Comment (WilliamJamieson) [inline romancal/skymatch/skystatistics.py:134]: This is no longer necessary, all it is doing is attaching the units that were stripped off.

Comment (mairanteodoro) [reply romancal/skymatch/skystatistics.py]: Thanks, William! I updated the code to remove that check.

File: pyproject.toml [hunk 3]
```diff
@@ -24,9 +24,9 @@ dependencies = [
     "pyparsing >=2.4.7",
     "requests >=2.26",
     # "rad>=0.21.0,<0.22.0",
-    "rad @ git+https://github.com/spacetelescope/rad.git",
+    "rad @ git+https://github.com/mairanteodoro/rad.git@RCAL-932-remove-units-in-exposure-level-pipeline",
```
Comment (PaulHuwe) [inline pyproject.toml:27]:
```suggestion
    "rad @ git+https://github.com/spacetelescope/rad.git",
```

File: pyproject.toml [hunk 4]
```diff
@@ -26,7 +26,7 @@ dependencies = [
     # "rad>=0.21.0,<0.22.0",
     "rad @ git+https://github.com/spacetelescope/rad.git",
     # "roman_datamodels>=0.21.0,<0.22.0",
-    "roman_datamodels @ git+https://github.com/spacetelescope/roman_datamodels.git",
+    "roman_datamodels @ git+https://github.com/mairanteodoro/roman_datamodels.git@RCAL-932-remove-units-in-exposure-level-pipeline",
```
Comment (PaulHuwe) [inline pyproject.toml:29]:
```suggestion
    "roman_datamodels @ git+https://github.com/spacetelescope/roman_datamodels.git",
```

**Review by schlafly — APPROVED** [review]

**Pull request merged**